### PR TITLE
Updates for TypeScript 1.6

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: modules/angular2/src/core/dom/dom_adapter.ts [hunk 1]
```diff
@@ -9,137 +8,129 @@ export function setRootDomAdapter(adapter: DomAdapter) {
...
-
-function _abstract() {
-  return new BaseException('This method is abstract');
-}
-
 /* tslint:disable:requireParameterType */
 /**
  * Provides DOM operations in an environment-agnostic way.
  */
 export class DomAdapter {
```
Comment (vicb) [inline modules/angular2/src/core/dom/dom_adapter.ts]: Should it be abstract (or a comment added if abstract is not supported yet)

Comment (alexeagle) [inline modules/angular2/src/core/dom/dom_adapter.ts]: Oops! good catch.

**Comment by alexeagle** [comment]
cc @tbosch

File: modules/angular2/manual_typings/traceur-runtime.d.ts [hunk 2]
```diff
@@ -7,13 +7,6 @@
 // ES5, because they are redundant with lib.es6.d.ts.
 /// <reference path="../typings/es6-promise/es6-promise.d.ts"/>
 
-// es6-promise.d.ts chose a different name for this interface than TS lib.es6.d.ts
```
Comment (vsavkin) [inline modules/angular2/manual_typings/traceur-runtime.d.ts]: Are you planning to rename this file?

File: modules/angular2/src/core/dom/parse5_adapter.ts [hunk 3]
```diff
@@ -562,6 +562,11 @@ export class Parse5DomAdapter extends DomAdapter {
   getAnimationPrefix(): string { return ''; }
   getTransitionEnd(): string { return 'transitionend'; }
   supportsAnimation(): boolean { return true; }
+
+  replaceChild(el, newNode, oldNode) { throw 'not implemented'; }
```
Comment (vsavkin) [inline modules/angular2/src/core/dom/parse5_adapter.ts]: could you throw new Error("not implemented")?

As far as know, JS does not keep the stack if you throw a simple string.

File: modules/angular2/manual_typings/traceur-runtime.d.ts (hunk 2 above)
Comment (alexeagle) [inline modules/angular2/manual_typings/traceur-runtime.d.ts]: yes, soon

File: modules/angular2/src/core/dom/parse5_adapter.ts (hunk 3 above)
Comment (alexeagle) [inline modules/angular2/src/core/dom/parse5_adapter.ts]: yup

**Pull request closed without merging**